metricflow/model/validations/measures.py

```diff
@@ -263,10 +273,54 @@ def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  #
                             ),
                             message=(
                                 f"Measure '{measure.name}' has a non_additive_dimension with name '{non_additive_dimension.name}' "
-                                f"that is defined as a categorical dimension which is not supported."
+                                f"that is not defined as a dimension in data source '{data_source.name}'."
                             ),
                         )
                     )
+                if matching_dimension:
+                    # Check that it's a time dimension
+                    if matching_dimension.type != DimensionType.TIME:
+                        issues.append(
+                            ValidationError(
+                                context=DataSourceElementContext(
+                                    file_context=FileContext.from_metadata(metadata=data_source.metadata),
+                                    data_source_element=DataSourceElementReference(
+                                        data_source_name=data_source.name, element_name=measure.name
+                                    ),
+                                    element_type=DataSourceElementType.MEASURE,
+                                ),
+                                message=(
+                                    f"Measure '{measure.name}' has a non_additive_dimension with name '{non_additive_dimension.name}' "
+                                    f"that is defined as a categorical dimension which is not supported."
+                                ),
+                            )
+                        )
+
+                    # Validates that the non_additive_dimension time_granularity is >= agg_time_dimension time_granularity
+                    if (
+                        matching_dimension.type_params
+                        and agg_time_dimension.type_params
+                        and (
+                            matching_dimension.type_params.time_granularity.name
+                            != agg_time_dimension.type_params.time_granularity.name
```

Comment:
Is this meaningfully different from `matching_dimension.type_params.time_granularity != agg_time_dimension.type_params.time_granularity`? I think it amounts to the same thing, in which case I'd prefer the simpler direct comparison.

Reply:
ah yes, i added that there as a test due to a bug I encountered and forgot to take it out! thanks!